pyoracc/test/atf/test_atflexer.py

```diff
@@ -896,6 +909,40 @@ def test_note_ended_by_strucuture(self):
             ["REVERSE"]
         )
 
+    def compare_note_ended_by_line(self, line_label):
+        'Helper for Note para state termination.'
+        # Sample text.
+        line1 = u"a-šar _saḫar.ḫi.a_ bu-bu-su-nu"
+        line2 = u"a-kal-ši-na ṭi-id-di"
+        # Generate the successive line numbers in the same style.
+        label1 = line_label
+        next = int(label1[:1]) + 1
+        if _pyversion() == 2:
+            label2 = unicode(next) + label1[1:]
+        else:
+            label2 = str(next) + label1[1:]
+        self.compare_tokens(
+            label1 + ". " + line1 + "\n" +
+            "#note: Does this combine with the next line?\n" +
+            label2 + ". " + line2 + "\n",
+            ["LINELABEL"] + ["ID"] * len(line1.split()) + ["NEWLINE"] +
+            ["NOTE", "ID", "NEWLINE"] +
+            ["LINELABEL"] + ["ID"] * len(line2.split()) + ["NEWLINE"],
+            [label1] + line1.split() +
+            [None, None, "Does this combine with the next line?", None] +
+            [label2] + line2.split() + [None]
+        )
+
+    def test_note_ended_by_line(self):
+        'Notes can be free text until the next line label.'
+        for label in ["1",
+                      "2'",
+                      u"3\u2019",
+                      u"4\u2032",
+                      u"5\u02CA",
+                      u"6\xb4"]:
+            self.compare_note_ended_by_line(label)
```

Comment:
Instead of a helper function, I think that it would be a bit clearer to parametrise the test (essentially the code in `compare_note_ended_by_line`) by the label. I may be missing something, though!

Reply:
It would. I tried to use `@pytest.mark.parametrize` here like in [`test_atffile.py`](https://github.com/oracc/pyoracc/blob/af363ab/pyoracc/test/atf/test_atffile.py#L93-L100), but couldn't get it to interact properly with the tests being class methods.

It looked like the `unittest` functionality could be replaced by a pytest fixture, moving all the tests up to module level and saving an indent level, but that's something which touches most of the file, so unless I'm missing something I thought it was better to leave that for a separate change.